__tests__/main.test.ts

```diff
@@ -157,4 +219,75 @@ describe('action', () => {
     expect(runMock).toHaveReturned()
     expect(getTokenMock).toHaveBeenCalled()
   })
+
+  it('runs a test suite, branch already exists', async () => {
+    getInputMock.mockImplementation(testSuiteInput)
+    getBooleanInputMock.mockReturnValue(false)
+
+    const branchID = uuid.v4()
+    const projectID = uuid.v4()
+    const systemID = uuid.v4()
+    const buildID = uuid.v4()
+    const batchID = uuid.v4()
+    const testSuiteID = uuid.v4()
+
+    getTokenMock.mockImplementation(async (): Promise<string> => {
+      return Promise.resolve('token')
+    })
+
+    getProjectIDMock.mockResolvedValueOnce(projectID)
+    getSystemIDMock.mockResolvedValueOnce(systemID)
+    getTestSuiteIDMock.mockResolvedValueOnce(testSuiteID)
+
+    // pretend we're in a PR to test this code path
+    process.env.GITHUB_HEAD_REF = 'pr-branch'
+    process.env.GITHUB_EVENT_NAME = 'pull_request'
+    process.env.GITHUB_ACTOR = 'github-user'
+    const associatedAccount: string = process.env.GITHUB_ACTOR
+    findOrCreateBranchMock.mockResolvedValueOnce(branchID)
+
+    Object.defineProperty(github, 'context', {
+      value: {
+        eventName: 'pull_request',
+        payload: {
+          pull_request: {
+            head: {
+              sha: '03403a4f2db7fc85c79c4da80bd3ea719eb8dce5'
+            },
+            number: 123,
+            title: 'Test PR'
+          }
+        }
+      }
+    })
+
+    const newBuild: Build = {
+      branchID,
+      systemID,
+      description: `#123 - Test PR`,
+      imageUri: 'a.docker/image:tag',
+      buildID,
+      version: '03403a4f'
+    }
+    createBuildMock.mockResolvedValueOnce(newBuild)
+
+    const newBatch: Batch = {
+      associatedAccount,
+      batchID
+    }
+
+    const testSuiteBatchInputMock = jest
+      .spyOn(BatchesApi.prototype, 'createBatchForTestSuite')
+      .mockResolvedValueOnce({ data: newBatch } as AxiosResponse)
+
+    await main.run()
+
+    expect(getProjectIDMock).toHaveBeenCalledTimes(1)
+    expect(getSystemIDMock).toHaveBeenCalledTimes(1)
+    expect(getTestSuiteIDMock).toHaveBeenCalledTimes(1)
+    expect(createBranchMock).not.toHaveBeenCalled()
```

Comment:
Once?

Reply:
this one is not called, but the batch is one time too, i can add that